Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,467 @@
+# coding=utf-8
+"""
+Written by Egbert Bouman, Mihai Capotă, Elric Milon, and Ardhi Putra Pratama H
+Supported boosting sources
+"""
+import HTMLParser
+import glob
+import logging
+import os
+import random
+import time
+import urllib
+from binascii import hexlify
+from hashlib import sha1
+
+import libtorrent as lt
+from twisted.internet import reactor
+from twisted.internet.task import LoopingCall
+from twisted.web.client import Agent, readBody
+from twisted.web.http_headers import Headers
+
+from Tribler.Core.TorrentDef import TorrentDef
+from Tribler.Core.simpledefs import NTFY_INSERT, NTFY_TORRENTS, NTFY_UPDATE
+from Tribler.Core.version import version_id
+from Tribler.Main.Utility.GuiDBHandler import startWorker
+from Tribler.Main.Utility.GuiDBTuples import Torrent
+from Tribler.Main.vwxGUI.GuiUtility import GUIUtility
+from Tribler.community.allchannel.community import AllChannelCommunity
+from Tribler.community.channel.community import ChannelCommunity
+from Tribler.dispersy.exception import CommunityNotFoundException
+from Tribler.dispersy.util import call_on_reactor_thread
+
+
+class BoostingSource(object):
+    """
+    Base class for boosting source. For now, it can be RSS, directory, and channel
+    """
+
+    def __init__(self, session, source, boost_settings, callback):
+        self.session = session
+        self.channelcast_db = session.lm.channelcast_db
+
+        self.torrents = {}
+        self.source = source
+        self.interval = boost_settings.source_interval
+        self.max_torrents = boost_settings.max_torrents_per_source
+        self.callback = callback
+        self.archive = False
+
+        self.enabled = True
+
+        self.av_uprate = 0
+        self.av_dwnrate = 0
+        self.storage_used = 0
+        self.ready = False
+
+        self.gui_util = GUIUtility.getInstance()
```

Comment:
We should try to avoid the use of `Manager` classes declared in the `vwxGUI` package since they will most likely be completely removed when we are going to nuke devel. Most of these "managers" are managing GUI-related objects.

For instance, you are using a `TorrentManager` and in particular, the `getTorrentByInfohash` method. This method just fetches a torrent from the database and returns it (look it up in the source code). You don't really need to use the `TorrentManager` object to do this, instead, you can directly invoke the `getTorrent` method in `TorrentDBHandler`.

@whirm do you agree on this? Not making use of any of the stuff declared in `vwxGUI` when we avoid it makes my life a lot easier when I'm going to remove the wx GUI.